test/integration/test_fake_backends.py

```diff
@@ -0,0 +1,712 @@
+            2: Qubit(QuantumRegister(3, "qr1"), 2),
+            3: Qubit(QuantumRegister(2, "qr2"), 0),
+            4: Qubit(QuantumRegister(2, "qr2"), 1),
+            5: Qubit(QuantumRegister(15, "ancilla"), 0),
+            6: Qubit(QuantumRegister(15, "ancilla"), 1),
+            7: Qubit(QuantumRegister(15, "ancilla"), 2),
+            8: Qubit(QuantumRegister(15, "ancilla"), 3),
+            9: Qubit(QuantumRegister(15, "ancilla"), 4),
+            10: Qubit(QuantumRegister(15, "ancilla"), 5),
+            11: Qubit(QuantumRegister(15, "ancilla"), 6),
+            12: Qubit(QuantumRegister(15, "ancilla"), 7),
+            13: Qubit(QuantumRegister(15, "ancilla"), 8),
+            14: Qubit(QuantumRegister(15, "ancilla"), 9),
+            15: Qubit(QuantumRegister(15, "ancilla"), 10),
+            16: Qubit(QuantumRegister(15, "ancilla"), 11),
+            17: Qubit(QuantumRegister(15, "ancilla"), 12),
+            18: Qubit(QuantumRegister(15, "ancilla"), 13),
+            19: Qubit(QuantumRegister(15, "ancilla"), 14),
+        }
+
+        vf2_layout = {
+            0: Qubit(QuantumRegister(15, "ancilla"), 0),
+            1: Qubit(QuantumRegister(15, "ancilla"), 1),
+            2: Qubit(QuantumRegister(15, "ancilla"), 2),
+            3: Qubit(QuantumRegister(15, "ancilla"), 3),
+            4: Qubit(QuantumRegister(15, "ancilla"), 4),
+            5: Qubit(QuantumRegister(15, "ancilla"), 5),
+            6: Qubit(QuantumRegister(3, "qr1"), 1),
+            7: Qubit(QuantumRegister(15, "ancilla"), 6),
+            8: Qubit(QuantumRegister(15, "ancilla"), 7),
+            9: Qubit(QuantumRegister(15, "ancilla"), 8),
+            10: Qubit(QuantumRegister(3, "qr1"), 0),
+            11: Qubit(QuantumRegister(3, "qr1"), 2),
+            12: Qubit(QuantumRegister(15, "ancilla"), 9),
+            13: Qubit(QuantumRegister(15, "ancilla"), 10),
+            14: Qubit(QuantumRegister(15, "ancilla"), 11),
+            15: Qubit(QuantumRegister(15, "ancilla"), 12),
+            16: Qubit(QuantumRegister(2, "qr2"), 0),
+            17: Qubit(QuantumRegister(2, "qr2"), 1),
+            18: Qubit(QuantumRegister(15, "ancilla"), 13),
+            19: Qubit(QuantumRegister(15, "ancilla"), 14),
+        }
+        # Trivial layout
+        expected_layout_level0 = trivial_layout
+        # Dense layout
+        expected_layout_level1 = vf2_layout
+        # CSP layout
+        expected_layout_level2 = vf2_layout
+        expected_layout_level3 = vf2_layout
+
+        expected_layouts = [
+            expected_layout_level0,
+            expected_layout_level1,
+            expected_layout_level2,
+            expected_layout_level3,
+        ]
+        backend = FakeTokyo()
+        result = transpile(qc, backend, optimization_level=level, seed_transpiler=42)
+        self.assertEqual(result._layout.initial_layout._p2v, expected_layouts[level])
+
+    @data(0, 1, 2, 3)
+    def test_layout_tokyo_fully_connected_cx(self, level):
```

Comment:
Thanks for taking care of this! Only minor comment I have is that this test is failing: 

```
AssertionError: {6: Qubit(QuantumRegister(5, 'qr'), 0), 5:[838 chars] 14)} != {0: Qubit(QuantumRegister(15, 'ancilla'), [838 chars] 14)}
Diff is 1699 characters long. Set self.maxDiff to None to see it
```

Reply:
Thanks for the review [kt474](https://github.com/kt474). After talking to Matthew he pointed out that there were better alternatives for integration tests, so I have ended up completely replacing this file. This test assumes that the current terra `main` is used (there are a few changes that are ahead of the release). Let me know if there are any issues with the new test too.